File: Java/src/com/company/动态数组/IArrayList.java
package com.company.动态数组;

public interface IArrayList {

    void clear();

    int size();

    boolean isEmpty();

    boolean contains(int element);

    void add(int element);

    int get(int index);

    int set(int index,int element);

    int add(int index,int element);

    int remove(int index);

    int indexOf(int element);
}

File: grade_one/动态数组/IArrayList.cs
namespace 动态数组
{
    /// <summary>
    /// 动态数组接口 Dynamic Array
    /// </summary>
    public interface IArrayList<T>
    {
        int Size();

        bool IsEmpty();

        bool Contains(T element);

        void Add(T element);

        void Add(T element,int index);

        T Remove(int index);

        void Clear();

        void Set(T element);

        void Set(T element, int index);

        T Get(int index);
        
        int IndexOf(T element);
    }
}